src/Components/Components/src/OwningComponent.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// A base class that creates a service provider scope, and resolves a service of type <typeparamref name="TService"/>.
+    /// </summary>
+    /// <typeparam name="TService">The service type.</typeparam>
+    /// <remarks>
+    /// Use the <see cref="OwningComponent{TService}"/> class as a base class to author components that control
+    /// the lifetime of a service or multiple services. This is useful when using a transient or scoped service that
+    /// requires disposal such as a repository or database abstraction. Using <see cref="OwningComponent{TService}"/>
+    /// as a base class ensures that the service and relates services that share its scope are disposed with the component.
+    /// </remarks>
+    public abstract class OwningComponent<TService> : ComponentBase, IDisposable
```

Comment:
What are your thoughts on calling this `OwningComponentBase`? We already have `ComponentBase` and `LayoutComponentBase`.

Reply:
I'm a little conflicted because the actual naming guidelines say to *not* call something FooBase - but we're already doing it and I don't really want to change it. So I guess I'll be consistent 👍